java-checks/src/main/java/org/sonar/java/checks/CallSuperMethodFromInnerClassCheck.java

```diff
@@ -79,7 +79,8 @@ public MethodInvocationVisitor(Symbol.TypeSymbol classSymbol) {
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
       Symbol symbol = tree.symbol();
-      if (symbol.isMethodSymbol() && tree.methodSelect().is(Tree.Kind.IDENTIFIER) && isInherited(symbol) && outerClassHasMethodWithSameName(symbol)) {
+      if (symbol.isMethodSymbol() && !"<init>".equals(symbol.name()) && tree.methodSelect().is(Tree.Kind.IDENTIFIER) && isInherited(symbol)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)